### ISSUE TITLE
`break` with label generates invalid code when iteration is transpiled

### ISSUE THREAD

**Issue by mischnic** [swc-project/swc#3366]
### Describe the bug

When using a for-of loop with arrow expressions and some nested for loops with variable declarations (see example), using break with a label causes invalid code because the break isn't inside the loop with that label anymore

### Input code

```typescript
scan: for (let x; foo; ) {
  foo.find((m) => true);
  for (let f of foo) {
  }
  x = foo();
  for (;;) {
    foo.find(() => true);
    if (foo) break scan;
  }
}
```

### Config

```json
{
  "jsc": {
    "parser": {
      "syntax": "ecmascript",
      "jsx": false
    },
    "target": "es2022",
    "loose": false,
    "minify": {
      "compress": false,
      "mangle": false
    }
  },
  "module": {
    "type": "es6"
  },
  "env": {
    "coreJs": 3,
    "mode": "usage"
  },
  "minify": false,
  "isModule": true
}
```

### Playground link

https://play.swc.rs/?version=1.2.133&code=H4sIAAAAAAAAA1WOQQqAMAwE775ij%2BnFBxj0L1UbKGoLrYIg%2Ft22CuohsOwOQ%2BKgXQPxATSbFTun7BkKR4Uca7FuJFoU2g5r2IziMjy8wEvGbv5Mt6PNBb0c871%2BfX8dYAVUNH0wekJMX3ERntUFhE3D%2BaIAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1FsWFLnKEQg8hXMW4%2BIdkl4bgu9dKE7oTM2%2BeNniJhXmDgizEesmaKn5gBrIRxbIvFaaBjWjBINQnqMiOqiJiLsaMOuQsdAATRJ%2F8sqrM5liYRP4VJhdOsg9XzM%2BmwQZ1LbQ7bzBySu%2BfgOk%2B5ted1L4JOlLi%2FHKYvTwOVeVG%2FQtn0eQh2gAAAA%3D%3D

### Expected behavior

The break with a label should somehow still work correctly

### Actual behavior

The break isn't nested in the for loop with that label anymore, so this is invalid code (and Terser correctly throws an error when minifying thsi)

```js
var _loop = function(x1) {
    // ...
    for(;;){
        foo.find(function() {
            return true;
        });
        if (foo) break scan;  // <--------
    }
    x = x1, void 0;
};
scan: for(var x; foo;)_loop(x); // <--------
```

### Version

1.2.133

### Additional context

https://github.com/parcel-bundler/parcel/issues/7585

**Comment by magic-akari** [comment]
fixed in #3360. It works well in 1.2.134 and 1.2.135.

**Comment by kdy1** [comment]
Thank you!
